test/SILOptimizer/mem2reg_ossa.sil

```diff
@@ -0,0 +1,485 @@
+// RUN: %target-sil-opt -enable-sil-verify-all %s -mem2reg | %FileCheck %s
```

Comment:
Again, it would probably be good to add some non-trivial tests.

Reply:
Can you add that.

Reply:
Also, if you think this is really ready for ossa, I wonder what the effect would be if we ran this before semantic arc opts? Before we do that, I am probably going to need to move the eager specializer though to get a nice effect (a bunch of allocations are needed due to in_guaranteed args.

Reply:
For sure. I wanted to make sure that this patch was going in the right direction before writing hundreds of lines of tests that would need to be re-written if I had to significantly change the direction of this patch.